Would anyone be able to restart the AppVeyor build, unless @peterschrammel is going make further changes anyway?

AppVeyor said:
```
C:\projects\cbmc\src\goto-instrument\cover_instrument_mcdc.cpp : fatal error C1083: Cannot open compiler generated file: 'cover_instrument_mcdc.obj': Permission denied
```
This looks like a transient (and inexplicable) error to me.

--- a/src/solvers/flattening/boolbv_equality.cpp
+++ b/src/solvers/flattening/boolbv_equality.cpp
@@ -12,6 +12,7 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include <util/std_expr.h>
 #include <util/base_type.h>
+#include <util/invariant.h>
 
 #include <langapi/language_util.h>
 
@@ -19,12 +20,12 @@ Author: Daniel Kroening, kroening@kroening.com
 
 literalt boolbvt::convert_equality(const equal_exprt &expr)
 {
-  if(!base_type_eq(expr.lhs().type(), expr.rhs().type(), ns))
-  {
-    std::cout << "######### lhs: " << expr.lhs().pretty() << '\n';
-    std::cout << "######### rhs: " << expr.rhs().pretty() << '\n';
-    throw "equality without matching types";
-  }
+  const bool is_base_type_eq =
+    base_type_eq(expr.lhs().type(), expr.rhs().type(), ns);
+  DATA_INVARIANT(
+    is_base_type_eq,
+    std::string("equality without matching types:\n") + "######### lhs: " +
+      expr.lhs().pretty() + '\n' + "######### rhs: " + expr.rhs().pretty());
 
   // see if it is an unbounded array
   if(is_unbounded_array(expr.lhs().type()))
@@ -68,12 +69,13 @@ literalt boolbvt::convert_verilog_case_equality(
   // This is 4-valued comparison, i.e., z===z, x===x etc.
   // The result is always Boolean.
 
-  if(!base_type_eq(expr.lhs().type(), expr.rhs().type(), ns))
-  {
-    std::cout << "######### lhs: " << expr.lhs().pretty() << '\n';
-    std::cout << "######### rhs: " << expr.rhs().pretty() << '\n';
-    throw "verilog_case_equality without matching types";
-  }
+  const bool is_base_type_eq =
+    base_type_eq(expr.lhs().type(), expr.rhs().type(), ns);
+  DATA_INVARIANT(
+    is_base_type_eq,
+    std::string("verilog_case_equality without matching types:\n") +
+      "######### lhs: " + expr.lhs().pretty() + '\n' +
+      "######### rhs: " + expr.rhs().pretty());
 
   const bvt &bv0=convert_bv(expr.lhs());
   const bvt &bv1=convert_bv(expr.rhs());

--- a/src/solvers/sat/cnf.cpp
+++ b/src/solvers/sat/cnf.cpp
@@ -12,9 +12,9 @@ Author: Daniel Kroening, kroening@kroening.com
 #include "cnf.h"
 
 #include <algorithm>
-#include <cassert>
-#include <iostream>
 #include <set>
+
+#include <util/invariant.h>
 
 // #define VERBOSE
 
@@ -426,10 +426,12 @@ bool cnft::process_clause(const bvt &bv, bvt &dest)
   for(const auto l : bv)
   {
     // we never use index 0
-    assert(l.var_no()!=0);
+    INVARIANT(l.var_no() != 0, "variable 0 must not be used");
 
     // we never use 'unused_var_no'
-    assert(l.var_no()!=literalt::unused_var_no());
+    INVARIANT(
+      l.var_no() != literalt::unused_var_no(),
+      "variable 'unused_var_no' must not be used");
 
     if(l.is_true())
       return true; // clause satisfied
@@ -437,11 +439,10 @@ bool cnft::process_clause(const bvt &bv, bvt &dest)
     if(l.is_false())
       continue; // will remove later
 
-    if(l.var_no()>=_no_variables)
-      std::cout << "l.var_no()=" << l.var_no()
-                << " _no_variables=" << _no_variables << '\n';
-
-    assert(l.var_no()<_no_variables);
+    INVARIANT(
+      l.var_no() < _no_variables,
+      "unknown variable " + std::to_string(l.var_no()) +
+        " (no_variables = " + std::to_string(_no_variables) + " )");
   }
 
   // now copy

--- a/src/solvers/sat/pbs_dimacs_cnf.cpp
+++ b/src/solvers/sat/pbs_dimacs_cnf.cpp
@@ -15,12 +15,14 @@ Author: Alex Groce
 
 void pbs_dimacs_cnft::write_dimacs_pb(std::ostream &out)
 {
-  double d_sum=0;
+  double d_sum = 0;
 
   // std::cout << "enter: No Lit.=" << no_variables () << "\n";
 
-  for(std::map<literalt, unsigned>::const_iterator it=pb_constraintmap.begin();
-      it!=pb_constraintmap.end(); ++it)
+  for(std::map<literalt, unsigned>::const_iterator it =
+        pb_constraintmap.begin();
+      it != pb_constraintmap.end();
+      ++it)
     d_sum += ((*it).second);
 
   if(!optimize)
@@ -45,7 +47,7 @@ void pbs_dimacs_cnft::write_dimacs_pb(std::ostream &out)
 
   for(const auto &lit_entry : pb_constraintmap)
   {
-    int dimacs_lit=lit_entry.first.dimacs();
+    int dimacs_lit = lit_entry.first.dimacs();
     out << "v" << dimacs_lit << " c" << lit_entry.second << "\n";
   }
 
@@ -61,7 +63,7 @@ bool pbs_dimacs_cnft::pbs_solve()
   if(!pbs_path.empty())
   {
     command += pbs_path;
-    if(command.substr(command.length(), 1)!="/")
+    if(command.substr(command.length(), 1) != "/")
       command += "/";
   }
 
@@ -69,20 +71,20 @@ bool pbs_dimacs_cnft::pbs_solve()
 
   // std::cout << "PBS COMMAND IS: " << command << "\n";
   /*
-    if (!(getenv("PBS_PATH")==NULL))
-    {
+  if (!(getenv("PBS_PATH")==NULL))
+  {
     command=getenv("PBS_PATH");
-    }
-    else
-    {
+  }
+  else
+  {
     error ("Unable to read PBS_PATH environment variable.\n");
     return false;
-    }
+  }
   */
 
   command += " -f temp.cnf";
 
-  #if 1
+#if 1
   if(optimize)
   {
     if(binary_search)
@@ -99,19 +101,19 @@ bool pbs_dimacs_cnft::pbs_solve()
   {
     command += " -S 1000 -D 1 -a";
   }
-  #else
+#else
   command += " -z";
-  #endif
+#endif
 
   command += " -a > temp.out";
 
-  int res=system(command.c_str());
-  CHECK_RETURN(0==res);
+  int res = system(command.c_str());
+  CHECK_RETURN(0 == res);
 
   std::ifstream file("temp.out");
   std::string line;
   int v;
-  bool satisfied=false;
+  bool satisfied = false;
 
   if(file.fail())
   {
@@ -119,60 +121,60 @@ bool pbs_dimacs_cnft::pbs_solve()
     return false;
   }
 
-  opt_sum=-1;
+  opt_sum = -1;
 
-  while(file && !file.eof ())
+  while(file && !file.eof())
+  {
+    std::getline(file, line);
+    if(
+      strstr(line.c_str(), "Variable Assignments Satisfying CNF Formula:") !=
+      nullptr)
     {
-      std::getline(file, line);
-      if(strstr(line.c_str(),
-                "Variable Assignments Satisfying CNF Formula:")!=nullptr)
+      // print ("Reading assignments...\n");
+      // std::cout << "No literals: " << no_variables() << "\n";
+      satisfied = true;
+      assigned.clear();
+      for(size_t i = 0; (file && (i < no_variables())); ++i)
+      {
+        file >> v;
+        if(v > 0)
         {
-          // print ("Reading assignments...\n");
-          // std::cout << "No literals: " << no_variables() << "\n";
-          satisfied=true;
-          assigned.clear();
-          for(size_t i=0; (file && (i < no_variables())); ++i)
-            {
-              file >> v;
-              if(v > 0)
-                {
-                  // std::cout << v << " ";
-                  assigned.insert(v);
-                }
-            }
-          // std::cout << "\n";
-          // print ("Finished reading assignments.\n");
+          // std::cout << v << " ";
+          assigned.insert(v);
         }
-      else if(strstr(line.c_str(), "SAT... SUM")!=nullptr)
-        {
-          // print (line);
-          sscanf(line.c_str(), "%*s %*s %*s %d", &opt_sum);
-        }
-      else if(strstr(line.c_str(), "SAT - All implied")!=nullptr)
-        {
-          // print (line);
-          sscanf(
-            line.c_str(),
-            "%*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %d",
-            &opt_sum);
-        }
-      else if(strstr(line.c_str(), "SAT... Solution")!=nullptr)
-        {
-          // print(line);
-          sscanf(line.c_str(), "%*s %*s %*s %d", &opt_sum);
-        }
-      else if(strstr(line.c_str(), "Optimal Soln")!=nullptr)
-        {
-          // print(line);
-          if(strstr(line.c_str(), "time out")!=nullptr)
-            {
-              status() << "WARNING:  TIMED OUT.  SOLUTION MAY BE INCORRECT."
-                       << eom;
-              return satisfied;
-            }
-          sscanf(line.c_str(), "%*s %*s %*s %d", &opt_sum);
-        }
+      }
+      // std::cout << "\n";
+      // print ("Finished reading assignments.\n");
     }
+    else if(strstr(line.c_str(), "SAT... SUM") != nullptr)
+    {
+      // print (line);
+      sscanf(line.c_str(), "%*s %*s %*s %d", &opt_sum);
+    }
+    else if(strstr(line.c_str(), "SAT - All implied") != nullptr)
+    {
+      // print (line);
+      sscanf(
+        line.c_str(),
+        "%*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %d",
+        &opt_sum);
+    }
+    else if(strstr(line.c_str(), "SAT... Solution") != nullptr)
+    {
+      // print(line);
+      sscanf(line.c_str(), "%*s %*s %*s %d", &opt_sum);
+    }
+    else if(strstr(line.c_str(), "Optimal Soln") != nullptr)
+    {
+      // print(line);
+      if(strstr(line.c_str(), "time out") != nullptr)
+      {
+        status() << "WARNING:  TIMED OUT.  SOLUTION MAY BE INCORRECT." << eom;
+        return satisfied;
+      }
+      sscanf(line.c_str(), "%*s %*s %*s %d", &opt_sum);
+    }
+  }
 
   return satisfied;
 }
@@ -191,21 +193,18 @@ propt::resultt pbs_dimacs_cnft::prop_solve()
   pbfile.close();
 
   // We start counting at 1, thus there is one variable fewer.
-  messaget::status() <<
-    (no_variables()-1) << " variables, " <<
-    clauses.size() << " clauses" << eom;
+  messaget::status() << (no_variables() - 1) << " variables, " << clauses.size()
+                     << " clauses" << eom;
 
-  bool result=pbs_solve();
+  bool result = pbs_solve();
 
   if(!result)
   {
-    messaget::status() <<
-      "PBS checker: system is UNSATISFIABLE" << eom;
+    messaget::status() << "PBS checker: system is UNSATISFIABLE" << eom;
   }
   else
   {
-    messaget::status() <<
-      "PBS checker: system is SATISFIABLE";
+    messaget::status() << "PBS checker: system is SATISFIABLE";
     if(optimize)
       messaget::status() << " (distance " << opt_sum << ")";
     messaget::status() << eom;
@@ -219,40 +218,40 @@ propt::resultt pbs_dimacs_cnft::prop_solve()
 
 tvt pbs_dimacs_cnft::l_get(literalt a) const
 {
-  int dimacs_lit=a.dimacs();
+  int dimacs_lit = a.dimacs();
 
   // std::cout << a << " / " << dimacs_lit << "=";
 
-  bool neg=(dimacs_lit < 0);
+  bool neg = (dimacs_lit < 0);
   if(neg)
-    dimacs_lit=-dimacs_lit;
+    dimacs_lit = -dimacs_lit;
 
-  std::set<int>::const_iterator f=assigned.find(dimacs_lit);
+  std::set<int>::const_iterator f = assigned.find(dimacs_lit);
 
   if(!neg)
+  {
+    if(f == assigned.end())
     {
-      if(f==assigned.end())
-        {
-          // std::cout << "FALSE" << "\n";
-          return tvt(false);
-        }
-      else
-        {
-          // std::cout << "TRUE" << "\n";
-          return tvt(true);
-        }
+      // std::cout << "FALSE" << "\n";
+      return tvt(false);
     }
+    else
+    {
+      // std::cout << "TRUE" << "\n";
+      return tvt(true);
+    }
+  }
   else
+  {
+    if(f != assigned.end())
     {
-      if(f!=assigned.end())
-        {
-          // std::cout << "FALSE" << "\n";
-          return tvt(false);
-        }
-      else
-        {
-          // std::cout << "TRUE" << "\n";
-          return tvt(true);
-        }
+      // std::cout << "FALSE" << "\n";
+      return tvt(false);
     }
+    else
+    {
+      // std::cout << "TRUE" << "\n";
+      return tvt(true);
+    }
+  }
 }

